api/api.go

```diff
@@ -0,0 +1,64 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/gorilla/context"
+	"github.com/mailgun/oxy/forward"
+)
+
+type (
+	// API services a set of HTTP routes
+	API struct {
```

Comment:
You can just have a `config` field and assign the passed config to it.  This removes the need to duplicate all of the fields from the Config struct.